### PR TITLE
Update custom section logic for schemas

### PR DESCRIPTION
**Pull request by schwartz-concordium**
## Purpose

Update custom section logic for schemas.

## Changes

Follow up on PR https://github.com/Concordium/concordium.github.io/pull/922

## Checklist

- [x] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: source/mainnet/smart-contracts/general/contract-schema.rst [hunk 1]
```diff
@@ -165,7 +165,12 @@ section`_ feature of Wasm modules.
 This allows Wasm modules to include a named section of bytes, which does not
 affect the semantics of running the Wasm module.
 
-All schemas are collected and added in one custom section named
-``concordium-schema``.
-This collection is a list of pairs, containing the name of the contract encoded
-in UTF-8 and the contract schema bytes.
+The module can contain a schema in one of two different custom sections. This collection is a list of pairs, containing the name of the contract encoded
+in UTF-8 and the contract schema bytes. The section 'concordium-schema' is the most common and is what the current tooling produces.
```
Comment (limemloh) [inline source/mainnet/smart-contracts/general/contract-schema.rst:169]:
```suggestion
in UTF-8 and the contract schema bytes. The section ``concordium-schema`` is the most common and is what the current tooling produces.
```

File: source/mainnet/smart-contracts/general/contract-schema.rst [hunk 2]
```diff
@@ -165,7 +165,12 @@ section`_ feature of Wasm modules.
...
+
+The supported sections depend on the module version. The schema version can be either defined by the section name or embedded into the actual schema:
+- Both 'V0' and 'V1' modules support the section 'concordium-schema' where the schema includes the version.
```
Comment (limemloh) [inline source/mainnet/smart-contracts/general/contract-schema.rst:172]:
```suggestion
- Both 'V0' and 'V1' modules support the section ``concordium-schema`` where the schema includes the version.
```

File: source/mainnet/smart-contracts/general/contract-schema.rst [hunk 3]
```diff
@@ -165,7 +165,12 @@ section`_ feature of Wasm modules.
...
+   - For 'V0' modules this is always a 'V0' schema.
+   - For 'V1' modules this can be a 'V1', 'V2', or 'V3' schema.
+- 'V0' modules additionally support section 'concordium-schema-v1' which always contain a 'V0' schema.
```
Comment (limemloh) [inline source/mainnet/smart-contracts/general/contract-schema.rst:175]:
```suggestion
- 'V0' modules additionally support section ``concordium-schema-v1`` which always contain a 'V0' schema.
```

File: source/mainnet/smart-contracts/general/contract-schema.rst [hunk 4]
```diff
@@ -165,7 +165,12 @@ section`_ feature of Wasm modules.
...
+- 'V1' modules additionally support section 'concordium-schema-v2' which always contain a 'V1' schema.
```
Comment (limemloh) [inline source/mainnet/smart-contracts/general/contract-schema.rst:176]:
```suggestion
- 'V1' modules additionally support section ``concordium-schema-v2`` which always contain a 'V1' schema.
```

**Review by limemloh — APPROVED** [review]
Great! I only have some nitpicking of the formatting.

**Review by dg-concordium — APPROVED** [review]

**Pull request merged**